pkg/util/archive/tar_xz.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package archive
+
+import (
+	"archive/tar"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/xi2/xz"
+)
+
+// ErrStopWalk can be returned in WalkTarXZArchive to stop walking
+var ErrStopWalk = errors.New("stop walk")
+
+// WalkTarXZArchive walks the provided .tar.xz archive, calling walkFunc for each entry.
+// If ErrStopWalk is returned from walkFunc, then the walk is stopped.
+func WalkTarXZArchive(tarxzArchive string, walkFunc func(*tar.Reader, *tar.Header) error) error {
+	f, err := os.Open(tarxzArchive)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	zr, err := xz.NewReader(f, 0)
+	if err != nil {
+		return err
+	}
+	tr := tar.NewReader(zr)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break // End of archive
+		}
+		if err != nil {
+			return err
+		}
+
+		if err := walkFunc(tr, hdr); err != nil {
+			if errors.Is(err, ErrStopWalk) {
+				return nil
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+// TarXZExtractFile extracts a single file at path from the provided archive
+func TarXZExtractFile(tarxzArchive string, path string, destinationDir string) error {
```

Comment:
@brycekahle Can/should this and https://github.com/DataDog/datadog-agent/blob/main/pkg/fleet/internal/tar/tar.go#L27 be somehow merged? May be lots of overlap.

Reply:
There is some overlap. That implementation supports directories and symlinks, whereas our current use cases do not need/want those.